src/modules/core/components/IndexModal/IndexModal.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import {
+  MessageDescriptor,
+  defineMessages,
+  FormattedMessage,
+} from 'react-intl';
+
+import Dialog, { DialogProps } from '~core/Dialog';
+import Heading from '~core/Heading';
+import styles from './IndexModal.css';
+import IndexModalItem from './IndexModalItem';
+import Button from '~core/Button';
+import Icon from '~core/Icon';
+
+const displayName = 'core.IndexModal';
+
+interface Items {
+  title: MessageDescriptor;
+  description: MessageDescriptor;
+  icon: string;
+  comingSoon?: boolean;
+  onClick?: () => void;
+}
+
+interface Props extends DialogProps {
+  items: Items[];
+  title: MessageDescriptor;
+  back?: () => void;
+}
+
+const MSG = defineMessages({
+  backButton: {
+    id: 'core.IndexModal.backButton',
+    defaultMessage: 'Back',
+  },
+});
+
+const IndexModal = ({ title, cancel, items, back }: Props) => {
+  return (
+    <Dialog cancel={cancel}>
+      <div className={styles.header}>
+        <Heading
+          appearance={{
+            margin: 'none',
+            size: 'medium',
+            weight: 'bold',
+            theme: 'dark',
+          }}
+          text={title}
+        />
+      </div>
+      <div className={styles.content}>
+        {items.map((item) => (
+          <IndexModalItem {...item} key={item.icon} />
+        ))}
+        {back && (
+          <Button
+            appearance={{ theme: 'secondary' }}
+            onClick={back}
+            className={styles.backButton}
+          >
+            <Icon
+              appearance={{ size: 'normal' }}
+              name="caret-left"
+              title={MSG.backButton}
+            />
+            <FormattedMessage {...MSG.backButton} />
```

Comment:
We have an id for this: https://github.com/JoinColony/colonyDapp/blob/e633b3b76d5e39ec4c6063f21dbdeb7c8de33392/src/i18n/en.json#L31

No need to declare another message descriptor